otx/algorithms/classification/tasks/inference.py

```diff
@@ -440,32 +438,9 @@ def _init_recipe_hparam(self) -> dict:
     def _init_recipe(self):
         logger.info("called _init_recipe()")
 
-        if self._multilabel:
-            recipe_root = os.path.join(MPAConstants.RECIPES_PATH, "stages/classification/multilabel")
-        else:
-            recipe_root = os.path.join(MPAConstants.RECIPES_PATH, "stages/classification")
-
         logger.info(f"train type = {self._train_type}")
 
-        if self._train_type in RECIPE_TRAIN_TYPE:
-            # TODO: this condition will be simplified after adding support for multlabel and hierarchical to SupCon.
-            if (
-                self._train_type == TrainType.INCREMENTAL
-                and not self._multilabel
-                and not self._hierarchical
-                and self._hyperparams.learning_parameters.enable_supcon
-                and not self._model_dir.endswith("supcon")
-            ):
-                recipe = os.path.join(recipe_root, "supcon.yaml")
-                self._model_dir = os.path.join(self._model_dir, "supcon")
-            else:
-                recipe = os.path.join(recipe_root, RECIPE_TRAIN_TYPE[self._train_type])
-        else:
-            raise NotImplementedError(f"Train type {self._train_type} is not implemented yet.")
-
-        logger.info(f"train type = {self._train_type} - loading {recipe}")
-
-        self._recipe_cfg = MPAConfig.fromfile(recipe)
+        self._recipe_cfg = self._init_model_cfg()
```

Comment:
How about changing function name `_init_model_cfg` to `_init_recipe` then?

Reply:
Please fix this in refactoring, it's too big a scope to cover in this PR.